Inquirer list prompt not selectable in newer versions (works in 12.10.0)

After upgrading Inquirer to versions newer than `12.10.0`, the interactive `list` prompt becomes unusable. Although the prompt message and choices are rendered correctly in the terminal, the user is unable to interact with them.

Specifically:

* Arrow keys do not move the selection
* Pressing Enter does not confirm any choice
* Keyboard input appears to be ignored entirely
* The cursor remains stuck on the prompt without any response

This completely blocks the CLI flow, since the user cannot proceed past the first question.

**Code Example**

```ts
const answers = await inquirer.prompt([
  {
    type: "list",
    name: "orm",
    message: "Choose Your ORM:",
    choices: ["Prisma", "Drizzle"],
  },
  {
    type: "list",
    name: "database",
    message: "Select Database:",
    choices: ["Postgresql", "Mongodb", "Mysql"],
    when: (ans) => ans.orm === "Prisma",
  },
  {
    type: "confirm",
    name: "authUi",
    message: "Do you want to include auth UI design?",
    default: true,
  },
]);
```

**Environment**

* OS: Windows 11
* Terminal: PowerShell
* Node.js: v22.18.0
* Inquirer version: Latest (issue occurs)
* Inquirer version: 12.10.0 (working correctly)

**Impact**
Any CLI relying on `inquirer.prompt` with `type: "list"` becomes unusable in this environment.

**Notes**

* No changes were made to the prompt logic
* Downgrading to `12.10.0` fully resolves the issue
* Likely a regression related to keyboard input handling or terminal compatibility in newer versions


**Code Reference**
The issue occurs in our actual CLI implementation here:
https://github.com/abdirahmanmahamoud/authverse/blob/main/package/authverse/cli/init.ts

**Update: Unable to test via gitpkg**

I tried installing the fix using the provided gitpkg URL, but it fails with a `402 Payment Required` error:

```bash
npm install https://gitpkg.now.sh/SBoudrias/Inquirer.js/packages/core?sboudrias/feat-research-couldve-introduced-18o
```

**Error:**

```
npm ERR! code E402
npm ERR! 402 Payment Required - GET https://gitpkg.vercel.app/...
```

It looks like `gitpkg` is no longer accessible or requires payment, so this method doesn’t work.

I also attempted installing directly via `pnpm` from GitHub, but the branch could not be resolved:

```bash
pnpm add @inquirer/core@github:SBoudrias/Inquirer.js#feat-research-couldve-introduced-18o
```

```
Could not resolve feat-research-couldve-introduced-18o to a commit
```

---

**Workaround**

I manually applied the suggested fix (moving `output.mute()` after `readline.createInterface()`), and it resolves the issue on Windows (PowerShell).

--- a/packages/core/src/lib/create-prompt.ts
+++ b/packages/core/src/lib/create-prompt.ts
@@ -54,17 +54,18 @@ export function createPrompt<Value, Config>(
     const output = new MuteStream();
     output.pipe(context.output ?? process.stdout);
 
-    // Pre-mute the output so that readline doesn't echo stale keystrokes
-    // to the terminal before the first render. ScreenManager will unmute/mute
-    // the output around each render call as needed.
-    output.mute();
-
     // oxlint-disable-next-line typescript/no-unsafe-type-assertion
     const rl = readline.createInterface({
       terminal: true,
       input,
       output,
     }) as unknown as InquirerReadline;
+
+    // Mute the output after readline has initialized so readline can perform
+    // any terminal setup writes (e.g. Windows Console API initialization)
+    // before suppressing output. ScreenManager will unmute/mute around each
+    // render call as needed.
+    output.mute();
     const screen = new ScreenManager(rl);
 
     const { promise, resolve, reject } = PromisePolyfill.withResolver<Value>();

